File: server/WatchStore.Domain/Entities/Order.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WatchStore.Domain.Entities
{
    public class Order
    {
        public int OrderId { get; set; }
        public int CustomerId { get; set; }
        public int PaymentId { get; set; }
        public string OrderStatus { get; set; }
        public decimal Total { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.Now;

        // Navigation properties
        public Customer Customer { get; set; }
        public Payment Payment { get; set; }
        public ICollection<OrderDetail> OrderDetails { get; set; }

        public Order()
        {
            OrderDetails = new HashSet<OrderDetail>();
        }
    }
}

File: client/src/types/order.ts
export type CreateOrder = {
  customerId: number;
  customerAddressId: number;
  paymentId: number;
  orderDetails: {
    productId: number;
    quantity: number;
  }[];
};

export type Order = {
  orderId: number;
  customerId: number;
  paymentId: number;
  orderStatus: string;
  total: number;
  createdAt: string;
};

export type OrderResponse = {
  orders: Order[];
  total: number;
  limit: number;
  skip: number;
};
